### ISSUE TITLE
Incorrect horizontal position of fixme warnings (W0511)

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4218]
The horizontal positions of the following four W0511 warning are all `2`:

`bug.py`:
```python
"""Horizontal position of fixme."""

# TODO 1
print(1)  # TODO 2


def fun():
    """Test in function."""
    # TODO 3
    if True & True:
        pass
        # TODO 4
```

`pylint bug.py`:
```
************* Module bug
bug.py:3:2: W0511: TODO 1 (fixme)
bug.py:4:2: W0511: TODO 2 (fixme)
bug.py:9:2: W0511: TODO 3 (fixme)
bug.py:12:2: W0511: TODO 4 (fixme)
```

This is how that is rendered by VS Code:
![image](https://user-images.githubusercontent.com/12128514/110423583-9bfbea00-80a1-11eb-8112-2261fa634032.png)

Shouldn't the position be the start of the comment (or the start of `TODO`)?

**Comment by Pierre-Sassoulas** [comment]
Thanks for the report. We added a column check to the functional test recently so it's probably just making sure that the proper number is used in fixme's functional tests and changing the naive value in the checker.

**Comment by bersbersbers** [comment]
I may be misunderstanding your comment, but just to be on the safe this: this is bug in a real-life application, not just a failing test.

**Comment by Pierre-Sassoulas** [comment]
Yes I was just talking about the way we can simply modify [the functional test text](https://github.com/PyCQA/pylint/blob/master/tests/functional/f/fixme.txt) and [the naive value in the checker](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/misc.py#L180) because of the amazing #4013 instead of having to create a new test :stuck_out_tongue:

**Comment by ksaketou** [comment]
Hello, I am a new contributor and I would like to work on this issue. :) I noticed that at `bug.py` if there is no space between `#` and `TODO`, the column number on the output will change to `1`, so the column number somehow depends on the position of  `#` for some reason. Do you think I should check the functional test for that or the checker itself?

**Comment by ksaketou** [comment]
Hello! I am a new contributor and I would like to work on this issue. I notice that for some reason the column index of the output depends on the position of  `#`. For example, if there is no space between `# `and `TODO`, the column number will change to 1, regardless of the column index of  `#`. Do you think  I should check the functional test for that or the checker itself?

**Comment by Pierre-Sassoulas** [comment]
Hello @ksaketou thank you for working on this, much appreciated ! I assigned you to the issue.

The fix need to be done on the checker itself but the column will change inside the test to reflect that :) You will be able to automatically update the expected output in the functional tests if the output is not the same anymore (`python tests/test_functional.py --update-functional-output -k "test_functional[fixme]"`) or you can change the value in the test manually first in order to check if your code is correct with `python3 -m pytest -k test_functional[fixme]`. The order is up to your personal preference :)

**Comment by ksaketou** [comment]
Great! I am working on it. I'll let you know. :)

**Comment by ksaketou** [comment]
Hello, I have a question. I made some changes on the checker and I successfully tested the code with `python -m pytest -k test_functional[fixme]`. However, when I use `python tests/test_functional.py --update-functional-output -k "test_functional[fixme]"` the column indexes at `fixme.txt` are updated with the wrong values. Why is this happening?

**Comment by Pierre-Sassoulas** [comment]
When you launch `python tests/test_functional.py --update-functional-output -k "test_functional[fixme]"` it updates the test with the current pylint output. You can use it to check the real output of your code. The problem here is that we're only testing columns for python 3.8 or above (the column information got different and better in python 3.8). So even if the column is wrong `python -m pytest -k test_functional[fixme]` will work everytime as long as the lines of the messages are correct. I think you might be using python 3.7 or below ? You will probably need to switch to python 3.8 for the test on column to be useful for you. You can develop under python 3.7 if you check the output yourself though. Then github actions will check all versions of python online.

**Comment by ksaketou** [comment]
Hello, sorry for the late reply. I see what you mean, thank you. Also, I use Python 3.8.1. I linked a pull request to this issue, please let me know if I need to make any changes.

**Comment by bersbersbers** [comment]
Thanks everyone!
